modules/dnn/src/onnx/onnx_importer.cpp

```diff
@@ -0,0 +1,582 @@
+        }
+        else if (attribute_proto.ints_size() > 0)
+        {
+            const ::google::protobuf::RepeatedField< ::google::protobuf::int64 >& src = attribute_proto.ints();
+            std::vector<int32_t> dst(attribute_proto.ints_size());
+
+            for (int j = 0; j < dst.size(); j++) {
+                if (src[j] < std::numeric_limits<int32_t>::min() || src[j] > std::numeric_limits<int32_t>::max()) {
+                    CV_Error(Error::StsOutOfRange, "Input is out of OpenCV 32S range");
+                }
+                dst[j] = saturate_cast<int32_t>(src[j]);
+            }
+
+                lp.set(attribute_proto.name(), DictValue::arrayInt(
+                    &dst[0], attribute_proto.ints_size()));
+        }
+        else if (attribute_proto.has_t())
+        {
+            ::opencv_onnx::TensorProto tensor = attribute_proto.t();
+            Mat blob = getMatFromTensor(tensor);
+            lp.blobs.push_back(blob);
+        }
+        else if (attribute_proto.has_t() || attribute_proto.has_g() ||
+                    attribute_proto.strings_size() > 0 ||
+                        attribute_proto.tensors_size() > 0 ||
+                            attribute_proto.graphs_size() > 0)
+        {
+                CV_Error(Error::StsNotImplemented, "Unexpected attribute type");
+        }
+        else
+            CV_Error(Error::StsNotImplemented, "Unsupported attribute type");
+    }
+    return lp;
+}
+
+Mat ONNXImporter::getBlob(const opencv_onnx::NodeProto& node_proto,
+                    const std::map<std::string, Mat>& constBlobs, int index)
+{
+    CV_Assert(index < node_proto.input_size());
+    std::map<std::string, Mat>::const_iterator constBlob;
+    constBlob = constBlobs.find(node_proto.input(index) );
+    if (constBlob == constBlobs.end()) {
+        CV_Error(Error::StsObjectNotFound,
+             "Blob " + node_proto.input(index) + " not found in const blobs");
+    }
+    return constBlob->second;
+}
+
+struct LayerInfo {
+    int layerId;
+    int outputId;
+    LayerInfo(int _layerId, int _outputId) : layerId(_layerId), outputId(_outputId) {}
+};
+
+void ONNXImporter::populateNet(Net dstNet)
+{
+    CV_Assert(model_proto.has_graph());
+    opencv_onnx::GraphProto graph_proto = model_proto.graph();
+    std::map<std::string, Mat> constBlobs = getGraphTensors(graph_proto);
+
+    std::string framework_name;
+    if (model_proto.has_producer_name()) {
+        framework_name = model_proto.producer_name();
+    }
+
+    // create map with network inputs (without const blobs)
+    std::map<std::string, LayerInfo> layer_id;
+    std::map<std::string, LayerInfo>::iterator layerId;
+    // fill map: push layer name, layer id and output id
+    int count = 0;
+    std::vector<String> netInputs;
+    for (int j = 0; j < graph_proto.input_size(); j++)
+    {
+        if (constBlobs.find(graph_proto.input(j).name()) == constBlobs.end()) {
```

Comment:
Please put `graph_proto.input(j).name())` to some temporal variable to remove duplications:
```
const std::string& name = graph_proto.input(j).name());
```

Reply:
done